mfr/extensions/image/static/js/jquery.zoom.js

```diff
@@ -0,0 +1,245 @@
+                }
+                if (!settings.url) {
+                    return;
+                }
+            }
+
+            $source.one("zoom.destroy", function(position, overflow) {
+                $source.off(".zoom");
+                target.style.position = position;
+                target.style.overflow = overflow;
+                img.onload = null;
+                $img.remove();
+            }.bind(this, target.style.position, target.style.overflow));
+
+            img.onload = function () {
+
+                var zoom = $.zoom(target, source, img, settings.magnify);
+
+                function start(e) {
+                    zoom.init();
+                    zoom.move(e);
+                    // Skip the fade-in for IE8 and lower. It chokes on fading-in and changing
+                    // position based on mousemovement at the same time.
+                    $img.stop().fadeTo(
+                        $.support.opacity ? settings.duration : 0, 1, $.isFunction(settings.onZoomIn) ? settings.onZoomIn.call(img) : false
+                    );
+                }
+
+                function stop() {
+                    $img.stop().fadeTo(
+                        settings.duration, 0, $.isFunction(settings.onZoomOut) ? settings.onZoomOut.call(img) : false
+                    );
+                }
+
+                // Mouse events
+                if (settings.on === "grab") {
+
+                    $source.on("mousedown.zoom", function (e) {
+                        if (e.which === 1) {
+                            $(document).one("mouseup.zoom", function () {
+                                stop();
+                                $(document).off(mousemove, zoom.move);
+                            });
+                            start(e);
+                            $(document).on(mousemove, zoom.move);
+                            e.preventDefault();
+                        }
+                    });
+
+                } else if (settings.on === "click") {
+
+                    // Added "mousewheel" event to enable further zoom when the images are enlarged
+                    $source.on("click.zoom", function (e) {
+                        if (!clicked) {
+                            clicked = true;
+                            start(e);
+                            $(document).on(mousemove, zoom.move);
+                            $(document).one("click.zoom",
+                                function () {
+                                    stop();
+                                    clicked = false;
+                                    $(document).off(mousemove, zoom.move);
+                                }
+                            );
+                            return false;
+                        }
+                        // do nothing if clicked is true, bubble the event up to the document to
+                        // trigger the unbind.
+                    }).on("mousewheel", function (e) {
```

Comment:
Added "mousewheel" event for our scroll zoom feature.

Reply:
I'm a little unclear on what this is doing and how this interacts with the mousewheel event handler you defined in the mako template.  Why `stop()` and `start()` here?  Why did you need to modify the `click.zoom` handler to remove the explicit return?  JS events aren't my strong point, but i think some more "why" comments would be helpful here.